### ISSUE TITLE
Implement completion guess for tabris

### ISSUE THREAD

**Issue by angelozerr** [angelozerr/tern-tabris#32]
Implement completion guess for tabris like:

* user write

```javascript
tabris.crea
```

* it does Ctrl+Space and completion shows create.
 * user apply completion for create
* a second popup completion opens with "Buttoné, "Page", etc

@cpetrov I think you could be interested since I have implemented completion guess https://github.com/angelozerr/tern-guess-types with CodeMirror. See demo at http://demo-angelozerr.rhcloud.com/CodeMirror-Java/guess-types.html of completion guess (it requires CodeMirror which is not released).

**Comment by cpetrov** [comment]
This is a great feature, thank you for the information! We will look into integrating it into tern-tabris.

**Comment by angelozerr** [comment]
Cool! So I will implement custom completion guess for tabris if you are interested.

**Comment by cpetrov** [comment]
@angelozerr Yes please. I can then remove the autocompletion empty string replacement workaround.

**Comment by angelozerr** [comment]
@cpetrov I'm wiating for CodeMirror release (>5.7.0) and I will implement it for tabris. You will see it's very easy. Do you think you will have time to continue my work once I will implement it for one tabris case (like tabris.create will show Page, etc when create is applyied?

**Comment by cpetrov** [comment]
@angelozerr Thank you! Yes, I can do that.

**Comment by angelozerr** [comment]
@cpetrov I have implemented completion guess for tabris.create & widget.get. Here the result:

![test](https://cloud.githubusercontent.com/assets/1932211/10354875/483c0d8c-6d65-11e5-80f1-9dae05c29ed5.gif)

Please note that we need to wait CodeMirror release https://github.com/codemirror/CodeMirror/issues/3573 in order to I create release of CodeMirror Extension+JavaSCript. For the moment I have pushed JS inside https://github.com/angelozerr/tern-tabris/tree/master/demos/scripts-waiting-for-cm-releases

You can if you wish implement completion guess ofr widget.set and event, etc Any feedback are welcome!

**Comment by cpetrov** [comment]
Thank you for the contribution @angelozerr!

**Comment by angelozerr** [comment]
@cpetrov CodeMirror has released 5.8.0 and I have released CodeMirror Extension+JavaScript. 

So now tern-tabris use thoses libraries.

Do you think we can create a release of tern-tabris?

**Comment by cpetrov** [comment]
@angelozerr Yes, thank you.
